contexts/authentication.tsx

```diff
@@ -125,14 +130,17 @@ export const AuthenticationProvider = ({ children }) => {
   useEffect(() => {
     if (!DAOService) return;
     
-    window?.ethereum?.on("accountsChanged", () => {
-      DAOService.connect()
-        .then(connected => {
-          if (connected) updateWalletAddress();
-        })
-        .catch(error => console.log("Failed to change account", error));
-    });
-  }, [DAOService]);
+    if (wallet?.address)
+      window?.ethereum?.on("accountsChanged", () => {
+        DAOService.connect()
+          .then(connected => {
+            if (connected) updateWalletAddress();
+          })
+          .catch(error => console.log("Failed to change account", error));
+      });
+    else
+      window?.ethereum?.removeAllListeners("accountsChanged");
```

Comment:
?!?!? what.. what if you connect the wallet again? xD

Reply:
@moshmage it will enter the if instead of else